src/utils/useFormWithValidation.js

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from "react";
+
+// hook for form control
+export function useForm() {
+  const [values, setValues] = React.useState({});
+
+  const handleChange = (event) => {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    setValues({ ...values, [name]: value });
+  };
+
+  return { values, handleChange, setValues };
+}
+
+// hook for form control and form validation
+export function useFormWithValidation() {
+  const [values, setValues] = React.useState({});
```

Comment:
Please, move `useForm` and `useFormWithValidation` into a `hooks` folder because theya are **hooks**